src/controllers/mainController.ts

```diff
@@ -321,13 +322,17 @@ export default class MainController implements vscode.Disposable {
      */
     private async createObjectExplorerSession(connectionCredentials?: IConnectionCredentials): Promise<void> {
         let createSessionPromise = new Deferred<TreeNodeInfo>();
-        await this._objectExplorerProvider.createSession(createSessionPromise, connectionCredentials);
-        const newNode = await createSessionPromise;
-        this._objectExplorerProvider.refresh(undefined);
-        let expandSessionPromise = new Deferred<TreeNodeInfo[]>();
-        await this._objectExplorerProvider.expandNode(newNode, newNode.sessionId, expandSessionPromise);
-        await expandSessionPromise;
-        this._objectExplorerProvider.refresh(undefined);
+        const sessionId = await this._objectExplorerProvider.createSession(createSessionPromise, connectionCredentials);
+        if (sessionId) {
+            const newNode = await createSessionPromise;
+            if (newNode) {
```

Comment:
what happens if no newNode?

Reply:
Nothing. There's no change so no need to refresh.